### ISSUE TITLE
git history ignores merge commits

### ISSUE THREAD

**Issue by tulinkry** [oracle/opengrok#1167]
Git history ignores the merge commits for particular files. When the merge commit did an actual change to the code then there is an inconsistent view on the file - file contains the correct current version but it's not visible which commit did the current version.

An example is file `src/org/opensolaris/opengrok/analysis/c/CXref.lex` in the OpenGrok.
The xref contains current:

```
120 <SCOMMENT> {
121 {WhiteSpace}*{EOL}      {
122     out.write(yytext());
123     yypop();
124     startNewLine();}
125 }
```

However according to the last history the content should be this:

```
119 <SCOMMENT> {
120 {WhiteSpace}*{EOL}      { yybegin(YYINITIAL); out.write("</span>");
121                  startNewLine();}
122 }
```

Merge commits are present in the git log command however they contain empty file list. Therefore they are not assigned to particular files and not displayed in the app.

Is there any intention in this?

**Comment by vladak** [comment]
Until `git log` can somehow display the filenames for merge commits, there is probably not much we can do about this.

**Comment by tulinkry** [comment]
Switches `-c --cc` kinda solves this for me, at least it seems like that.

**Comment by vladak** [comment]
Looks like `-c -cc` is for `git diff` only ?

**Comment by tulinkry** [comment]
Closed by #3167
